static/app/views/automations/components/editAutomationActions.tsx

```diff
@@ -0,0 +1,81 @@
+import {useCallback} from 'react';
+
+import {addSuccessMessage} from 'sentry/actionCreators/indicator';
+import {openConfirmModal} from 'sentry/components/confirm';
+import {Button} from 'sentry/components/core/button';
+import {ButtonBar} from 'sentry/components/core/button/buttonBar';
+import {t} from 'sentry/locale';
+import type {Automation} from 'sentry/types/workflowEngine/automations';
+import {useNavigate} from 'sentry/utils/useNavigate';
+import useOrganization from 'sentry/utils/useOrganization';
+import {
+  useDeleteAutomationMutation,
+  useUpdateAutomation,
+} from 'sentry/views/automations/hooks';
+import {makeAutomationBasePathname} from 'sentry/views/automations/pathnames';
+
+interface EditAutomationActionsProps {
+  automation: Automation;
+}
+
+export function EditAutomationActions({automation}: EditAutomationActionsProps) {
+  const organization = useOrganization();
+  const navigate = useNavigate();
+  const {mutate: deleteAutomation, isPending: isDeleting} = useDeleteAutomationMutation();
+  const {mutate: updateAutomation, isPending: isUpdating} = useUpdateAutomation();
+
+  const toggleDisabled = useCallback(() => {
+    const newEnabled = !automation.enabled;
+    updateAutomation(
+      {
+        automationId: automation.id,
+        ...automation,
+        enabled: newEnabled,
+      },
+      {
+        onSuccess: () => {
+          addSuccessMessage(
+            newEnabled ? t('Automation enabled') : t('Automation disabled')
+          );
+        },
+      }
+    );
+  }, [updateAutomation, automation]);
+
+  const handleDelete = useCallback(() => {
+    openConfirmModal({
+      message: t('Are you sure you want to delete this automation?'),
+      confirmText: t('Delete'),
+      priority: 'danger',
+      onConfirm: () => {
+        deleteAutomation(automation.id, {
```

Comment:
If you use the `mutateAsync` function returned from `useDeleteAutomationMutation()` and return the promise to `onConfirm` it will display a loading indicator in the confirm modal until the deletion is finished which is a better UX